pandas/tests/test_window.py

```diff
@@ -207,6 +207,26 @@ def f():
             'A', 'ra', 'std'), ('B', 'rb', 'mean'), ('B', 'rb', 'std')])
         tm.assert_frame_equal(result, expected, check_like=True)
 
+    def test_count_nonnumeric_types(self):
+        # GH12541
+        df_inf = DataFrame({'x': [1, 2, 3], 'y': [1., 2., np.Inf]})
+        df_date = DataFrame({'x': [1, 2, 3], 
+                            'y': pd.date_range('20130101',periods=3)})
```

Comment:
can you add some NaT in here for the datetimes,  add in a timedelta example, and a float example with np.nan.